source/guide_prosody.rst

```diff
@@ -0,0 +1,404 @@
+
+ [isabell@stardust ~]$ VERSION=0.11.5
+ [isabell@stardust ~]$ wget https://prosody.im/downloads/source/prosody-$VERSION.tar.gz --directory-prefix=$HOME/var/lib/prosody
+ [...]
+ [isabell@stardust ~]$ tar --extract --gzip --file=$HOME/var/lib/prosody/prosody-$VERSION.tar.gz --directory=$HOME/var/lib/prosody
+ [isabell@stardust ~]$ rm ~/var/lib/prosody/prosody-$VERSION.tar.gz
+ [isabell@stardust ~]$
+
+Configure, build and install prosody:
+
+::
+
+ [isabell@stardust ~]$ cd ~/var/lib/prosody/prosody-$VERSION
+ [isabell@stardust prosody-0.11.5]$ ./configure --ostype=linux --prefix=$HOME --with-lua-include=/usr/include
+ Lua version detected: 5.1
+ Lua interpreter found: /usr/bin/lua...
+ Checking Lua includes... lua.h found in /usr/include/lua.h
+ Checking if Lua header version matches that of the interpreter... yes
+ Writing configuration...
+
+ Installation prefix: /home/isabell
+ Prosody configuration directory: /home/isabell/etc/prosody
+ Using Lua from: /usr
+
+ Done. You can now run 'make' to build.
+
+ [isabell@stardust prosody-0.11.5]$ make
+ [...] 
+ [isabell@stardust prosody-0.11.5]$ make install
+ [...] 
+ [isabell@stardust prosody-0.11.5]$ 
+
+Configuration
+=============
+
+Generate SSL dhparam file
+-------------------------
+
+.. note:: This is going to take a long time!
+
+To improve the security you can generate a Diffie–Hellman parameter file with ``openssl``:
+
+::
+
+ [isabell@stardust ~]$ openssl dhparam -out ~/etc/prosody/certs/dhparam-4096.pem 4096
+ [...]
+ [isabell@stardust ~]$ 
+
+Install modules
+---------------
+
+Create the *http_upload* folder for the module **http_upload** which let clients upload files over HTTP. Additionally download the latest available community plugins:
+
+.. code-block:: console
+
+ [isabell@stardust ~]$ mkdir ~/var/lib/prosody/http_upload
+ [isabell@stardust ~]$ hg clone https://hg.prosody.im/prosody-modules/ ~/var/lib/prosody/prosody-modules
+ [isabell@stardust ~]$
+
+Configure prosody
+-----------------
+
+Then there are many settings which should be edited accordingly in ``~/etc/prosody/prosody.cfg.lua``. You'll find a explaination of the config file under the `example configuration file`_ from Prosody.
+
+Additionally I recommend the ssl ciphers and options to reach a high security score. You can check it over the `IM Observatory`_.
+
+.. code-block:: lua
+ :emphasize-lines: 2,3, 26, 29, 32, 38, 43-45, 50, 51, 53, 56, 57, 59
+
+ ---------- Server-wide settings ----------
+ admins = { "isabell@uber.space" }
+ plugin_paths = { "/home/isabell/var/lib/prosody/prosody-modules" }
+ modules_enabled = {
+   "roster";
+   "saslauth";
+   "tls";
+   "dialback";
+   "disco";
+   "carbons";
+   "pep";
+   "private";
+   "blocklist";
+   "vcard4";
+   "vcard_legacy";
+   "version";
+   "uptime";
+   "time";
+   "ping";
+   "register";
+   "mam";
+   "csi_simple";
+   "admin_adhoc";
+ allow_registration = false
+ c2s_require_encryption = true
+ c2s_ports = { C2S-PORT }
+ s2s_require_encryption = true
```

Comment:
This is already in the default config, no need to set them again. Please remove it to keep the guide shorter.